src/coreclr/System.Private.CoreLib/src/ILLink/ILLink.Descriptors.xml

```diff
@@ -25,6 +25,18 @@
     <type fullname="System.SZArrayHelper">
       <method name=".ctor" />
     </type>
+
+    <!-- Native hosting accesses managed methods in the ComponentActivator class. -->
+    <type fullname="Internal.Runtime.InteropServices.ComponentActivator">
```

Comment:
Why not make this conditional behind the feature switch as well? We should still keep the ifs in place (just in case something else calls it for example), but it should be perfectly fine to completely remove the entry points.